From 9aa135db21907fb7e3ae2aaccddd0418dc145afb Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 30 May 2013 16:14:58 +0530
Subject: HDMI disconnect during multichannel playback fails

BZ: 58732

The direct output thread exits on disconnect. Hence
on subsequent playback in IHF, the stream config will
be new. So new track/ device needs to be opened.
Therefore return dead object in these scenarios.

Category: device enablement
Domain: AUDIO_MEDIA-Audio Flinger
Origin: Internal
Upstream-Candidate: no

Change-Id: Ie8689f3b77abb433932f13e2b83be19a3abb11f6
Orig-Change-Id: I60c2534217a57ca70ba612537707152a887c185e
Signed-off-by: Jerome Anand <jerome.anand@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 services/audioflinger/Tracks.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/services/audioflinger/Tracks.cpp b/services/audioflinger/Tracks.cpp
index af04ce7..e433b5f 100644
--- a/services/audioflinger/Tracks.cpp
+++ b/services/audioflinger/Tracks.cpp
@@ -624,6 +624,10 @@ status_t AudioFlinger::PlaybackThread::Track::start(AudioSystem::sync_event_t ev
             buffer.mFrameCount = 1;
             (void) mAudioTrackServerProxy->obtainBuffer(&buffer, true /*ackFlush*/);
         }
+    } else if (thread == 0) {
+      /*in case of direct output thread, on disconnect; the thread exits hence
+       we need to make sure the tracks are restored */
+        status = DEAD_OBJECT;
     } else {
         status = BAD_VALUE;
     }
-- 
1.7.4.1

